src/fandango/evolution/population.py

```diff
@@ -20,30 +20,57 @@ def __init__(
         self.warnings_are_errors = warnings_are_errors
         self.max_nodes = max_nodes
 
+    @staticmethod
     def add_unique_individual(
-        self,
         population: List[DerivationTree],
         candidate: DerivationTree,
         unique_set: Set[int],
     ) -> bool:
+        """
+        Adds individual to the population if it is unique, according to its hash.
+
+        :param population: The population to potentially add the individual to.
+        :param candidate: The individual to potentially add to the population.
+        :param unique_set: The set of unique individuals.
+        :return: True if the individual was added, False otherwise.
+        """
         h = hash(candidate)
         if h not in unique_set:
             unique_set.add(h)
             population.append(candidate)
             return True
         return False
```

Comment:
I used this in a couple of more places that did the same thing.

Is there a reason we're not just using a simple set of `DerivationTree`s that is converted to a list when needed instead of a parallel list and set of hashes? Is it faster?